### PR TITLE
Remove downloading result from external url

### PR DESCRIPTION
**Pull request by kt474**
### Summary

From a slack conversation with @daka1510 regarding when `_download_external_result` was first added [here](https://github.com/Qiskit/qiskit-ibm-runtime/pull/709) to support osprey

> We have now cleaned up the corresponding code on the server-side to support this special result flow. Feel free to remove it from the SDK, too, if no one in this group objects.
Allowing clients to fetch data from COS directly via the SDK might become a priority again, but probably not follow this design with pre-signed URLs. Since it's all unclear right now, it's probably not a good idea to keep this code around.

### Details and comments
Fixes #

### CODE REVIEW

File: test/unit/test_circuit_runner_job.py [hunk 1]
```diff
@@ -34,6 +34,7 @@ def __init__(self, data: Any) -> None:
         """
         self._data = data
         api_client = MagicMock()
+        api_client.job_results = MagicMock(return_value=self._data)
```
Comment (ptristan3) [inline test/unit/test_circuit_runner_job.py:37]: Why is this line added? It seems to have no relation to the PR we want to revert

Comment (kt474) [reply test/unit/test_circuit_runner_job.py:37]: In `test_circuit_metadata_ndarray` we were mocking `_download_external_result` to just return the data given so since that method has been removed, i'm mocking `job_results()`

**Review by ptristan3 — APPROVED** [review]
